src/block/block_compact.c

```diff
@@ -65,6 +70,295 @@ __wt_block_compact_get_progress_stats(WT_SESSION_IMPL *session, WT_BM *bm,
+    }
+
+    if (!estimate)
+        __wt_verbose_level(session, WT_VERB_COMPACT, WT_VERBOSE_DEBUG_1,
+          "%s: total reviewed %" PRIu64 " pages, total rewritten %" PRIu64 " pages", block->name,
+          block->compact_pages_reviewed, block->compact_pages_rewritten);
+    __wt_verbose_level(session, WT_VERB_COMPACT,
+      (estimate ? WT_VERBOSE_DEBUG_3 : WT_VERBOSE_DEBUG_1),
+      "%s:%s %" PRIuMAX "MB (%" PRIuMAX ") available space in the first 80%% of the file",
+      block->name, estimate ? " estimating --" : "", (uintmax_t)avail_eighty / WT_MEGABYTE,
+      (uintmax_t)avail_eighty);
+    __wt_verbose_level(session, WT_VERB_COMPACT,
+      (estimate ? WT_VERBOSE_DEBUG_3 : WT_VERBOSE_DEBUG_1),
+      "%s:%s %" PRIuMAX "MB (%" PRIuMAX ") available space in the first 90%% of the file",
+      block->name, estimate ? " estimating --" : "", (uintmax_t)avail_ninety / WT_MEGABYTE,
+      (uintmax_t)avail_ninety);
+    __wt_verbose_level(session, WT_VERB_COMPACT,
+      (estimate ? WT_VERBOSE_DEBUG_3 : WT_VERBOSE_DEBUG_1),
+      "%s:%s require 10%% or %" PRIuMAX "MB (%" PRIuMAX
+      ") in the first 90%% of the file to perform compaction, compaction %s",
+      block->name, estimate ? " estimating --" : "", (uintmax_t)(file_size / 10) / WT_MEGABYTE,
+      (uintmax_t)(file_size / 10), *skipp ? "skipped" : "proceeding");
+}
+
+/*
+ * __block_compact_estimate_remaining_work --
+ *     Estimate how much more work the compaction needs to do for the given file.
+ */
+static void
+__block_compact_estimate_remaining_work(WT_SESSION_IMPL *session, WT_BLOCK *block)
+{
+    WT_EXT *ext;
+    wt_off_t avg_block_size, avg_internal_block_size, depth1_size, leaves_per_internal_page;
+    wt_off_t compact_start_off, extra_space, file_size, last, off, rewrite_size, size, write_off;
+    uint64_t n, pages_to_move, pages_to_move_orig, total_pages_to_move;
+    int compact_pct_tenths, iteration;
+    bool skip;
+
+    /*
+     * We must have reviewed at least some interesting number of pages for any estimates below to be
+     * worthwhile.
+     */
+    if (block->compact_pages_reviewed < WT_THOUSAND)
+        return;
+
+    /* Assume that we have already checked whether this file can be skipped. */
+    WT_ASSERT(session, block->compact_pct_tenths > 0);
+
+    /*
+     * Get the average block size that we encountered so far during compaction. Note that we are not
+     * currently accounting for overflow pages, as compact does not currently account for them
+     * either.
+     */
+    avg_block_size = (wt_off_t)WT_ALIGN(
+      block->compact_bytes_reviewed / block->compact_pages_reviewed, block->allocsize);
+
+    /* We don't currently have a way to track the internal page size, but this should be okay. */
+    avg_internal_block_size = 4096;
+    leaves_per_internal_page =
+      (wt_off_t)(block->compact_pages_reviewed / block->compact_internal_pages_reviewed);
+    depth1_size = avg_block_size * leaves_per_internal_page + avg_internal_block_size;
+
```

Comment:
It would be helpful to comment what you are calculating in depth1_size: e.g., "the size of one level of Btree, including the internal block and the leaf blocks reachable from that internal block". Or something like that. Or a more verbose/descriptive variable name could accomplish the clarify.